integration_test/tests/wallet.rs

```diff
@@ -151,6 +151,18 @@ fn wallet__dump_wallet() {
     }
 }
 
+#[test]
+fn wallet__encrypt_wallet() {
+    let node = Node::with_wallet(Wallet::None, &[]);
+
+    // NOTE: Create a new wallet and returns a dedicated client instance for it.
+    // This client will send RPCs to the `/wallet/test-wallet` endpoint.
+    let wallet_client = node.create_wallet("test-wallet").expect("createwallet");
+
+    // Use the dedicated client to encrypt the new wallet
+    let _: EncryptWallet = wallet_client.encrypt_wallet("test-passphrase").expect("encryptwallet");
+}
+
```

Comment:
```suggestion
#[test]
fn wallet__encrypt_wallet() {
    let node = Node::with_wallet(Wallet::Default, &[]);
    let _: EncryptWallet = node.client.encrypt_wallet("test-passphrase").expect("encryptwallet");
}
```

I looked into this and the reason for the feature gating on the above tests `dumpprivkey` and `dumpwallet`  is because they are only supported on legacy wallets, which is default on v22 and below, v23-25 with a dedicated function, and not possible above 25. So we need to keep the feature gating on tests themselves to work in v23-25.

The test here works on all versions and doesn't need the feature gating at all.

The current change works, but the whole test can just be as above.

Reply:
Interesting... Thank you for checking it out and for the suggestion.